com.gzoltar.cli/src/main/java/com/gzoltar/cli/commands/RunTestMethods.java

```diff
@@ -77,8 +81,15 @@ public int execute(final PrintStream out, final PrintStream err) throws Exceptio
       throw new RuntimeException(this.testMethods + " does not exist or cannot be read");
     }
 
-    final URL[] classpathURLs =
-        ((URLClassLoader) Thread.currentThread().getContextClassLoader()).getURLs();
+    List<URI> classpath = new ClassGraph().getClasspathURIs();
+    List<URL> classpathUrlList = new ArrayList();
+
+    for (URI temp : classpath) {
+	classpathUrlList.add(temp.toURL());
+    }
+
+    URL[] classpathURLs = new URL[classpathUrlList.size()];
+    classpathURLs = classpathUrlList.toArray(classpathURLs);
 
```

Comment:
Any reason to use `getClasspathURIs` instead of `getClasspathURLs`?  According to the [javadoc](https://javadoc.io/doc/io.github.classgraph/classgraph/latest/io/github/classgraph/ClassGraph.html) they sound very similar.  I looked at [ClassGraph's](https://github.com/classgraph/classgraph) source code and `getClasspathURLs` seems to call `getClasspathURIs` and then collect the URL of each URI (pretty much your code).  Assuming `getClasspathURLs` could be used, I guess this code could be a simple as
```java
final URL[] classpathURLs = new ClassGraph().getClasspathURLs().toArray(new URL[0]);
```

(Note: you can find more info in [here](https://stackoverflow.com/a/9572820/998816) on why it has been suggested to use an empty array as argument of the `toArray` function instead of a pre-sized array.)

Disclaimer: I'm not a [ClassGraph](https://github.com/classgraph/classgraph) expert.

Reply:
Thank you for the comment. My code was done in a hurry for my project. This looks nicer. I have tried it and it works good.